clash-lib/src/Clash/Core/Evaluator/Types.hs

```diff
@@ -70,12 +154,8 @@ type PrimUnwind
   -> Machine
   -> Maybe Machine
 
-type PrimEvaluator = (PrimStep, PrimUnwind)
-
 data Machine = Machine
-  { mPrimStep   :: PrimStep
-  , mPrimUnwind :: PrimUnwind
-  , mHeapPrim   :: PrimHeap
+  { mHeapPrim   :: PrimHeap
```

Comment:
Now that you've changed it, please add some documentation to this data type as wel.